src/routes/Profile/get-referrer-link.ts

```diff
@@ -0,0 +1,32 @@
+import axios from 'axios';
+
+export const getReferrerLink = async (account: string) => {
+  const baseUrl = process.env.REACT_APP_REFERRAL_AND_SIGNATURE_SERVICE_URL;
+  if (!baseUrl || !account) {
+    return undefined;
+  }
+
+  try {
+    const data = (
+      await axios.get<{
+        status?: string;
+        description?: string;
+        refers_with_code?: string;
+        wallet_address?: string;
+      }>(`${baseUrl}/get-refers-with/${account}`)
+    )?.data;
+
+    if (!data) {
+      return undefined;
+    }
+    const status = data.status ? parseInt(data.status, 10) : 200;
+    if (status !== 200 || !data.refers_with_code) {
+      return undefined;
+    }
+    const link = `${window.location.origin}/?invitedBy=${data.refers_with_code}`;
```

Comment:
A quick test suggests that this doesn't work with https://github.com/Voltz-Protocol/voltz-ui/pull/375, which was tested with URIs of the form http://localhost:3000/#/trader-pools?invitedBy=SWAEUlit cos that's what we used in the hackathon. If you think this URL without the `/#/trader-pools` part is better, then please can you suggest a new home for the localStorage-writing code in https://github.com/Voltz-Protocol/voltz-ui/pull/375 ?

Reply:
yes I will fix this in #375 and test it accordingly. Thanks for pointing it out.

Reply:
Great, thanks! Please resolve this conversation once fixed in #375